op-e2e/actions/l2_verifier.go

```diff
@@ -229,15 +234,30 @@ func (s *L2Verifier) ActL1FinalizedSignal(t Testing) {
 	s.finalizer.Finalize(t.Ctx(), finalized)
 }
 
+// syncStep represents the Driver.syncStep
+func (s *L2Verifier) syncStep(ctx context.Context) error {
+	if fcuCalled, err := s.engine.TryBackupUnsafeReorg(ctx); fcuCalled {
+		return err
+	}
+	if err := s.engine.TryUpdateEngine(ctx); !errors.Is(err, derive.ErrNoFCUNeeded) {
+		return err
```

Comment:
This will return `nil` if `err` is `nil`. Which is what we want but does take a bit of thinking about. Is this a common go idiom or should we be explicit with `err != nil && !errors.Is(err, derive.ErrNoFCUNeeded)`?

Same thing fro the `!= io.EOF` check below.

Reply:
It's indeed not clean code, but the way it worked previously. I'm going to file an issue to change this. I don't want to break it when already doing a large refactor.

Reply:
https://github.com/ethereum-optimism/optimism/issues/10725